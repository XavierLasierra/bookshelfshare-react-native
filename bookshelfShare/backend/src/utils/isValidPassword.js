const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
