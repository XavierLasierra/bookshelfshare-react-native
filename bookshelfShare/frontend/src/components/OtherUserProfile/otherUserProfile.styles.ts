import {StyleSheet} from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  socialContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 20,
  },
  follower: {
    fontSize: stylesConstants.fontSize.small,
    color: stylesConstants.colors.followDark,
    marginTop: 5,
  },
  activityIndicator: {
    marginTop: 50,
  },
});
