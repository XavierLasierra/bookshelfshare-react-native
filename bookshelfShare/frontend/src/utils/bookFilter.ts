export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
