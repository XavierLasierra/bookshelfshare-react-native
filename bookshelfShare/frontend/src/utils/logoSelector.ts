import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
