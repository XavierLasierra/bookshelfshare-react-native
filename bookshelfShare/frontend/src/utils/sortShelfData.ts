export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
