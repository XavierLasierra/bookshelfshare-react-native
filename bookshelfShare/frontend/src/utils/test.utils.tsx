import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
