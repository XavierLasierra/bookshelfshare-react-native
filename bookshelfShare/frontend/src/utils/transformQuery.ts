interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
