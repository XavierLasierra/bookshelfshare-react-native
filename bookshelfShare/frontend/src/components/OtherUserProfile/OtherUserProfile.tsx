/* eslint-disable no-underscore-dangle */
import React, { useEffect, useState } from 'react';
import {
  SafeAreaView, View, Image, Text, TouchableOpacity, ScrollView, ActivityIndicator
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';

import Header from '../Header/Header';
import ProfileDetail from '../ProfileDetail/ProfileDetail';

import { addUserFollowing, deleteUserFollowing } from '../../redux/actions/loggedUser.creator';
import { getBooksData } from '../../redux/actions/books.creator';
import { clearCurrentUser } from '../../redux/actions/currentUser.creator';
import logoSelector from '../../utils/logoSelector';

import RedProfileIcon from '../../assets/redProfileIcon.svg';
import AddUserIcon from '../../assets/addUserIcon.svg';
import globalStyles from '../../styles/global.styles';
import styles from './otherUserProfile.styles';
import stylesConstants from '../../styles/styles.constants';

export default function OtherUserProfile({ navigation, route: { params: { logo } } }: any) {
  const dispatch = useDispatch();
  const { userData: { _id: loggedUserId } } = useSelector((store: any) => store.loggedUser);
  const { following } = useSelector((store: any) => store.userSocials);
  const { token, refreshToken } = useSelector((store: any) => store.tokens);
  const { user, results } = useSelector((store: any) => store.currentUser);
  const [isDisabled, setIsDisabled] = useState(false);

  const isFollowed = following.some((followedUser: any) => followedUser._id === user._id);
  const isFollower = user?.following
  && user.following.some(({ _id: followingId }: any) => followingId === loggedUserId);

  useEffect(() => {
    setIsDisabled(false);
  }, [following]);

  useEffect(() => () => {
    dispatch(clearCurrentUser());
  }, []);

  function handleBookResultsPage(listName: string, bookArray: string[]) {
    dispatch(getBooksData(bookArray, token, refreshToken));
    navigation.push('BookResults',
      {
        listName: `${user.username}'s\n${listName}`,
        logo
      });
  }

  function handleAddFollowing() {
    dispatch(addUserFollowing(user._id, loggedUserId, token, refreshToken));
    setIsDisabled(true);
  }

  function handleDeleteFollowing() {
    dispatch(deleteUserFollowing(user._id, loggedUserId, token, refreshToken));
    setIsDisabled(true);
  }

  const currentUserDetail = (
    <View style={globalStyles.profileContainer}>
      <View style={globalStyles.topContainer}>
        <Image
          source={{ uri: user?.photo }}
          style={globalStyles.userPhoto}
        />
        <View style={globalStyles.userInformationContainer}>
          <Text style={globalStyles.userUsername}>{user?.username}</Text>
          <Text style={globalStyles.userEmail}>{user?.email}</Text>
          {isFollower && <Text style={styles.follower}>Follows you</Text>}
        </View>
      </View>
      <View>
        <View style={styles.socialContainer}>
          {isFollowed
            ? (
              <TouchableOpacity
                onPress={handleDeleteFollowing}
                disabled={isDisabled}
                testID="deleteFollowingButton"
              >
                <RedProfileIcon width={35} height={35} />
              </TouchableOpacity>
            )
            : (
              <TouchableOpacity
                onPress={handleAddFollowing}
                disabled={isDisabled}
                testID="addFollowingButton"
              >
                <AddUserIcon width={35} height={35} />
              </TouchableOpacity>
            )}
        </View>
        <ProfileDetail
          handleBookResultsPage={handleBookResultsPage}
          userBooks={user.books}
          shelves={[]}
        />
      </View>
    </View>
  );

  return (
    <SafeAreaView style={globalStyles.mainContainer}>
      <Header Logo={logoSelector(logo)} BackButton navigation={navigation} />
      <ScrollView>
        {results
          ? currentUserDetail
          : <ActivityIndicator size="large" color={stylesConstants.colors.dark} style={styles.activityIndicator} />}
      </ScrollView>
    </SafeAreaView>
  );
}
