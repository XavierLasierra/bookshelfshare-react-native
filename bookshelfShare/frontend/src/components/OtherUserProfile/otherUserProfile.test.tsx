import React from 'react';
import OtherUserProfile from './OtherUserProfile';
import {render, fireEvent} from '../../utils/test.utils';
import loggedUserMock from '../../mocks/loggedUser.mock';
import userSocialsMock from '../../mocks/userSocials.mock';
import currentUserMock from '../../mocks/currentUser.mock';
import {
  addUserFollowing,
  deleteUserFollowing,
} from '../../redux/actions/loggedUser.creator';
import {getBooksData} from '../../redux/actions/books.creator';
import loggedUserActions from '../../redux/actions/loggedUser.actions';
import navigationMock from '../../mocks/navigation.mock';

jest.mock('../../redux/actions/loggedUser.creator');
jest.mock('../../redux/actions/books.creator');

describe('Given a OtherUserProfile component', () => {
  describe('When it is rendered', () => {
    const route = {
      params: {
        logo: '',
      },
    };
    describe('And currentUser is loading', () => {
      test('Then should match the snapshot', () => {
        const initialState = {
          loggedUser: {userData: loggedUserMock.user},
        };
        const screen = render(
          <OtherUserProfile navigation={navigationMock} route={route} />,
          initialState,
        );
        expect(screen).toMatchSnapshot();
      });
    });

    describe('And currentUser is not followed / following you', () => {
      let screen: any;
      let navigation: any;
      beforeEach(() => {
        (addUserFollowing as jest.Mock).mockReturnValue({
          type: loggedUserActions.UPDATE_USER_FOLLOWING,
          data: [...userSocialsMock.following, {_id: '2'}],
        });
        (getBooksData as jest.Mock).mockReturnValue({type: ''});
        navigation = {
          ...navigationMock,
          push: jest.fn(),
        };

        const initialState = {
          loggedUser: {userData: loggedUserMock.user},
          userSocials: userSocialsMock,
          currentUser: currentUserMock[0],
        };
        screen = render(
          <OtherUserProfile navigation={navigation} route={route} />,
          initialState,
        );
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });
      describe('And you click on addFollowingButton', () => {
        beforeEach(() => {
          const addFollowingButton = screen.getByTestId('addFollowingButton');
          fireEvent.press(addFollowingButton);
        });

        test('Then addUserFollowing should have been called', () => {
          expect(addUserFollowing).toHaveBeenCalled();
        });

        test('Then a button with testId deleteFollowingButton should be rendered', () => {
          expect(screen.queryByTestId('deleteFollowingButton')).not.toBe(null);
        });
      });

      describe('And you click in one of the books lists', () => {
        beforeEach(() => {
          const bookListButton = screen.getByTestId('readButton');
          fireEvent.press(bookListButton);
        });
        test('Then getBooksData should have been called', () => {
          expect(getBooksData).toHaveBeenCalled();
        });
        test('Then navigation.push should have been called', () => {
          expect(navigation.push).toHaveBeenCalled();
        });
      });
    });

    describe('And currentUser is followed / following you', () => {
      let screen: any;
      beforeEach(() => {
        (deleteUserFollowing as jest.Mock).mockReturnValue({
          type: loggedUserActions.UPDATE_USER_FOLLOWING,
          data: userSocialsMock.following.filter(({_id}) => _id === '3'),
        });
        const initialState = {
          loggedUser: {userData: loggedUserMock.user},
          userSocials: userSocialsMock,
          currentUser: currentUserMock[1],
        };
        screen = render(
          <OtherUserProfile navigation={navigationMock} route={route} />,
          initialState,
        );
      });
      test('Then should match the snapshot', () => {
        expect(screen).toMatchSnapshot();
      });
      describe('And you click on deleteFollowingButton', () => {
        test('Then deleteUserFollowing should have been called', () => {
          const deleteFollowingButton = screen.getByTestId(
            'deleteFollowingButton',
          );
          fireEvent.press(deleteFollowingButton);

          expect(deleteUserFollowing).toHaveBeenCalled();
        });
      });
    });
  });
});
