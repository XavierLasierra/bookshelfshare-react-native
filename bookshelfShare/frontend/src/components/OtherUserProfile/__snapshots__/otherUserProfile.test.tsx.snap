// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a OtherUserProfile component When it is rendered And currentUser is followed / following you Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "photo",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 125,
                "marginRight": 30,
                "width": 125,
              }
            }
          />
          <View
            style={
              Object {
                "maxWidth": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                }
              }
            >
              User2
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              user2@gmail.com
            </Text>
            <Text
              style={
                Object {
                  "color": "blue",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              Follows you
            </Text>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 20,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="deleteFollowingButton"
            >
              <IconMock
                height={35}
                width={35}
              />
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readingButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              currently reading
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="toReadButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              to read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="wishlistButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              on wishlist
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
          >
            <IconMock
              height={35}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              shelves
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Given a OtherUserProfile component When it is rendered And currentUser is loading Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <ActivityIndicator
        color="#000000"
        size="large"
        style={
          Object {
            "marginTop": 50,
          }
        }
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Given a OtherUserProfile component When it is rendered And currentUser is not followed / following you Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "photo",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 125,
                "marginRight": 30,
                "width": 125,
              }
            }
          />
          <View
            style={
              Object {
                "maxWidth": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                }
              }
            >
              User1
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              user1@gmail.com
            </Text>
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 20,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="addFollowingButton"
            >
              <IconMock
                height={35}
                width={35}
              />
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readingButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              currently reading
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="readButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="toReadButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              to read
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
            testID="wishlistButton"
          >
            <IconMock
              height={40}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              books
               
              on wishlist
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 2,
                "borderColor": "#ECECEC",
                "borderTopWidth": 2,
                "flexDirection": "row",
                "height": 50,
                "marginBottom": 15,
                "opacity": 1,
              }
            }
          >
            <IconMock
              height={35}
              width={40}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "marginLeft": 15,
                }
              }
            >
              0
               
              shelves
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
